File: python/9-1.py
import math
import random

N = 10000

def print_result( sa, sb, i ):
    ave = sa / i
    hh = sb - i * ave * ave
    sig = math.sqrt( hh/ i )
    print( f"{i:6d} {ave:8.4f} {sig:8.4f}" )

def gauss():
    ss = 0.0

    for i in range(12):
        ss += random.random()
    return ss - 6.0

sa = 0.0
sb = 0.0
print( "回数\t平均値\t標準偏差" )
for i in range( 1, N+1 ):
    v = gauss()
    sa += v
    sb += v * v
    if( ( i % 1000 ) == 0 ):
        print_result( sa, sb, i )

File: julia/9-1.jl
# 正規乱数検証計算

using Printf

const N = 10000     # 繰り返し回数

# 結果打出し
function print_result( sa, sb, i )
    ave = sa / i
    hh = sb - i * ave^2
    σ = sqrt( hh / i )
    @printf( "%6d %8.4f %8.4f\n", i, ave, σ )
end

# 正規乱数生成
function gauss()
    ss = 0.0
    for i=1:12
        ss += rand()
    end
    return ss - 6.0
end

sa = 0.0
sb = 0.0
println( "回転\t平均値\t標準偏差" )     # 見出し打出し
for i=1:N
    v = gauss()
    global sa += v
    global sb += v^2
    if i%1000 == 0
        print_result( sa, sb, i )
    end
end
